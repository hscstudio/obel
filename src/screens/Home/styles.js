import styled from '@emotion/styled/macro'
import tw from 'tailwind.macro'

export const Main = styled.main`
  min-height: 800px;
  ${tw`
    flex flex-col items-center justify-center
    text-white bg-gray-800
    text-3xl
  `}
`

export const Logo = styled.img`
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
  @keyframes App-logo-spin {
    from {
      transform: rotate(0deg);
    }
    to {
      transform: rotate(360deg);
    }
  }
`

export const Link = tw.a`
  text-blue-500
  hover:underline
`
