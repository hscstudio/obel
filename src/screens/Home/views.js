import objstr from 'obj-str'
import PropTypes from 'prop-types'
import React from 'react'
import logo from './logo.svg'

export const Button = ({ children, on, onToggle }) => (
  <button
    className={
      'hover:underline ' +
      objstr({
        'text-blue-400': !on,
        'text-green-400': on
      })
    }
    onClick={() => onToggle()}
  >
    {children}
  </button>
)

Button.propTypes = {
  children: PropTypes.node.isRequired,
  on: PropTypes.bool.isRequired,
  onToggle: PropTypes.func.isRequired
}

export const Logo = () => <img alt='logo' className='spin h-64' src={logo} />
