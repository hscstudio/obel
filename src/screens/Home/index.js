import React, { useState } from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faMotorcycle } from '@fortawesome/pro-light-svg-icons'
import objstr from 'obj-str'
import logo from './logo.svg'
import { Main, Logo } from './styles'

function Home() {
  const [toggle, setToggle] = useState(false)
  return (
    <div className="text-center">
      <Main>
        <Logo alt="logo" src={logo} />
        <p>
          Ojek <FontAwesomeIcon icon={faMotorcycle} /> Belanja
        </p>
        <button
          className={
            'hover:underline ' +
            objstr({
              'text-blue-400': !toggle,
              'text-green-400': toggle,
            })
          }
          onClick={e => {
            e.preventDefault()
            setToggle(!toggle)
          }}
        >
          Learn React
        </button>
      </Main>
    </div>
  )
}

export default Home
