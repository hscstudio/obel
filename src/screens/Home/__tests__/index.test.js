import React from 'react'
import { render, fireEvent } from '@testing-library/react'

import Home from '..'

it('renders without crashing', () => {
  const { getByText } = render(<Home />)
  const title = getByText(/ojek.+belanja/i)
  expect(title).toBeDefined()

  const link = getByText(/learn react/i)
  expect(link).toMatchSnapshot()

  fireEvent.click(link)
  expect(link).toMatchSnapshot()
})
