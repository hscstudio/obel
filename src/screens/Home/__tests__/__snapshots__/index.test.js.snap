// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<button
  class="hover:underline text-blue-400"
>
  Learn React
</button>
`;

exports[`renders without crashing 2`] = `
<button
  class="hover:underline text-green-400"
>
  Learn React
</button>
`;
